### PR TITLE
Changes to support latest release of VIA, moved files to "idb" folder in "keymaps" in preparation of adding more boards, made changes to LAYOUT_ macros

### PR DESCRIPTION
**Pull request by itspngu**
## Description

* Changes to support latest release of VIA
* Moved files to "idb" folder in "keymaps" in preparation of adding more boards
* Made changes to LAYOUT_ macros

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by itspngu** [comment]
@Wilba6582

**Comment by itspngu** [comment]
It seems that during my attempts to get things to compile the LUFA stuff somehow ended up in here. I'm not a very avid git user, so if you could provide me with instructions on how to get that out of here without losing my ability to compile keymaps I'd appreciate it.

File: keyboards/idb/idb_60/config.h [hunk 1]
```diff
@@ -48,7 +48,8 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
 /* COL2ROW or ROW2COL */
 #define DIODE_DIRECTION COL2ROW
 
-/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+/* Debounce reduces chatter (unintended double-presses) - see https://beta.docs.qmk.fm/features/feature_debounce_type for more information */
```
Comment (fauxpark) [inline keyboards/idb/idb_60/config.h]:
```suggestion
/* Debounce reduces chatter (unintended double-presses) - see https://docs.qmk.fm/#/feature_debounce_type for more information */
```

File: keyboards/idb/idb_60/config.h [hunk 2]
```diff
@@ -20,10 +20,10 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID    0x6060 // 24672
-#define PRODUCT_ID   0x6060 // 24672
-#define DEVICE_VER   0x0001
-#define MANUFACTURER    pngu
+#define VENDOR_ID    	0x494b // idb Keyboards
```
Comment (fauxpark) [inline keyboards/idb/idb_60/config.h]:
```suggestion
#define VENDOR_ID    	0x494B // idb Keyboards
```

File: keyboards/idb/idb_60/keymaps/all_keys/keymap.c [hunk 3]
```diff
@@ -2,26 +2,41 @@
...
+	_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+	_______, _______, _______,                                     _______,                            _______, _______, _______
+  ),
 
 };
 
 // Layer LED indicators
 
 layer_state_t layer_state_set_user(layer_state_t state) {
+
```
Comment (fauxpark) [inline keyboards/idb/idb_60/keymaps/all_keys/keymap.c]:
```suggestion
```

File: keyboards/idb/idb_60/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	_______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+	_______,          _______,                                     _______,                            _______,          _______
+  ),
+
+};
+
+// Layer LED indicators
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+
```
Comment (fauxpark) [inline keyboards/idb/idb_60/keymaps/default/keymap.c]:
```suggestion
```

File: keyboards/idb/idb_60/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	_______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+	_______,          _______,                                     _______,                            _______,          _______
+  ),
+
+};
+
+// Layer LED indicators
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+
```
Comment (fauxpark) [inline keyboards/idb/idb_60/keymaps/via/keymap.c]:
```suggestion
```

File: keyboards/idb/idb_60/keymaps/via/rules.mk [hunk 6]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+LINK_TIME_OPTIMIZATION_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/idb/idb_60/keymaps/via/rules.mk]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/idb/idb_60/readme.md [hunk 7]
```diff
@@ -8,10 +8,19 @@ Hardware Availability: [/u/omgitspngu](https://www.reddit.com/user/omgitspngu/)
...
+    make idb/idb_60:default
 
 Or to make and flash:
 
-    make idb_60:default:dfu
+    make idb/idb_60:default:dfu
+	
+To compile VIA firmware:
+
+	make idb/idb_60:via
```
Comment (fauxpark) [inline keyboards/idb/idb_60/readme.md]:
```suggestion
    make idb/idb_60:via
```

File: keyboards/idb/idb_60/readme.md [hunk 8]
```diff
@@ -8,10 +8,19 @@ Hardware Availability: [/u/omgitspngu](https://www.reddit.com/user/omgitspngu/)
...
+	
+LAYOUT_* Macros:
+
+	LAYOUT_default: ANSI WKL with full Backspace
```
Comment (fauxpark) [inline keyboards/idb/idb_60/readme.md]:
```suggestion
 * *LAYOUT_default*: ANSI WKL with full Backspace
```

File: keyboards/idb/idb_60/readme.md [hunk 9]
```diff
@@ -8,10 +8,19 @@ Hardware Availability: [/u/omgitspngu](https://www.reddit.com/user/omgitspngu/)
...
+	LAYOUT_all: All split keys are available
```
Comment (fauxpark) [inline keyboards/idb/idb_60/readme.md]:
```suggestion
 * *LAYOUT_all*: All split keys are available
```

File: keyboards/idb/idb_60/rules.mk [hunk 10]
```diff
@@ -0,0 +1,30 @@
+# MCU name
+
```
Comment (fauxpark) [inline keyboards/idb/idb_60/rules.mk]:
```suggestion
```

File: keyboards/idb/idb_60/rules.mk [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+
```
Comment (fauxpark) [inline keyboards/idb/idb_60/rules.mk]:
```suggestion
```

File: keyboards/idb/idb_60/rules.mk [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+BOOTLOADER = atmel-dfu
+
+# Build Options
+# Comment out to disable the options.
+
```
Comment (fauxpark) [inline keyboards/idb/idb_60/rules.mk]:
```suggestion
```

File: keyboards/idb/idb_60/rules.mk [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+BOOTMAGIC_ENABLE = lite
+MOUSEKEY_ENABLE = yes			# Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/idb/idb_60/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = yes			# Mouse keys
```

File: keyboards/idb/idb_60/rules.mk [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+EXTRAKEY_ENABLE = yes			# Audio control and System control(+450)
```
Comment (fauxpark) [inline keyboards/idb/idb_60/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes			# Audio control and System control
```

File: keyboards/idb/idb_60/rules.mk [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+CONSOLE_ENABLE = no				# Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/idb/idb_60/rules.mk]:
```suggestion
CONSOLE_ENABLE = no				# Console for debug
```

**Comment by noroadsleft** [comment]
> It seems that during my attempts to get things to compile the LUFA stuff somehow ended up in here. I'm not a very avid git user, so if you could provide me with instructions on how to get that out of here without losing my ability to compile keymaps I'd appreciate it.

Somewhere along the line you modified your LUFA submodule and it's out of sync with QMK's. Before you fix this, you'll want to apply the suggested changes @fauxpark gave you, because it'll mean they're done and you don't potentially have to apply them again later.

----

Right now your `idb_60_via` branch points at 0f0b91429d1d346fd969f72729f9df1f1e0874fe, which is partially based on your modified LUFA submodule. To fix this, open your CLI terminal within your qmk_firmware repository and run:

```
git remote -v
```

You should see one name that is assigned to your fork of QMK, and one that points to QMK's repo. Mine looks like this:

```
$ git remote -v
origin  https://github.com/noroadsleft/qmk_firmware.git (fetch)
origin  https://github.com/noroadsleft/qmk_firmware.git (push)
upstream        https://github.com/qmk/qmk_firmware.git (fetch)
upstream        https://github.com/qmk/qmk_firmware.git (push)
```

Note which name points to each repo you have configured. Whichever name is assigned to QMK's repo, you'll want to fetch that master branch, then do an interactive rebase of your branch onto the QMK `master` branch. This will be something like:

```
git fetch upstream master && git rebase -i FETCH_HEAD
```

When you run the above command, you should get a window similar to this:

image

This is how Vim appears in my terminal. You may end up with a different editor, in which case these instructions fall apart. I only know Vim for this part. :slightly_frowning_face: If you *do* get a Vim window, we're okay here.

Type `dd`. This will delete the first line, that has the `added lufa` commit. After that, type `:wq` and hit Enter. This will save and exit your Vim window, and put you back into the terminal. Git will do some processing, then say something like `Successfully rebased and updated idb_60_via.`

From here, run `git push --force-with-lease` and your PR will update to a changeset without a modified LUFA submodule.

----

At some point, you'll need to get your `master` branch sorted as well. There are instructions on fixing this here:

- [Resynchronizing a Branch - QMK Firmware Docs](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch)

The ideal way to use and update your fork is detailed here:

- [Using Your Fork's Master - QMK Firmware Docs](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

If you need any help with this let me know.

File: keyboards/idb/idb_60/config.h (hunk 1 above)
Comment (itspngu) [reply keyboards/idb/idb_60/config.h]: updated according to requested changes

File: keyboards/idb/idb_60/config.h (hunk 2 above)
Comment (itspngu) [reply keyboards/idb/idb_60/config.h]: updated according to requested changes

File: keyboards/idb/idb_60/keymaps/all_keys/keymap.c (hunk 3 above)
Comment (itspngu) [reply keyboards/idb/idb_60/keymaps/all_keys/keymap.c]: updated according to requested changes

File: keyboards/idb/idb_60/keymaps/default/keymap.c (hunk 4 above)
Comment (itspngu) [reply keyboards/idb/idb_60/keymaps/default/keymap.c]: updated according to requested changes

File: keyboards/idb/idb_60/keymaps/via/keymap.c (hunk 5 above)
Comment (itspngu) [reply keyboards/idb/idb_60/keymaps/via/keymap.c]: updated according to requested changes

File: keyboards/idb/idb_60/rules.mk (hunk 11 above)
Comment (itspngu) [reply keyboards/idb/idb_60/rules.mk]: updated according to requested changes

File: keyboards/idb/idb_60/rules.mk (hunk 12 above)
Comment (itspngu) [reply keyboards/idb/idb_60/rules.mk]: updated according to requested changes

File: keyboards/idb/idb_60/rules.mk (hunk 13 above)
Comment (itspngu) [reply keyboards/idb/idb_60/rules.mk]: updated according to requested changes

File: keyboards/idb/idb_60/rules.mk (hunk 14 above)
Comment (itspngu) [reply keyboards/idb/idb_60/rules.mk]: updated according to requested changes

File: keyboards/idb/idb_60/rules.mk (hunk 15 above)
Comment (itspngu) [reply keyboards/idb/idb_60/rules.mk]: updated according to requested changes

**Comment by itspngu** [comment]
Thank you for the review and the help with git. Hoping everything is in a mergeable state now.

File: keyboards/idb/idb_60/config.h [hunk 16]
```diff
@@ -20,10 +20,10 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID    0x6060 // 24672
-#define PRODUCT_ID   0x6060 // 24672
-#define DEVICE_VER   0x0001
-#define MANUFACTURER    pngu
+#define VENDOR_ID    	0x494B // idb Keyboards
```
Comment (noroadsleft) [inline keyboards/idb/idb_60/config.h]: Tiny note here.

```suggestion
#define VENDOR_ID    	0x494B // "IB" for idb Keyboards
```

File: keyboards/idb/idb_60/idb_60.h [hunk 17]
```diff
@@ -36,3 +36,24 @@ inline void _idb_60_esc_led_off(void) {
...
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16        	  }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26           }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36           }, \
+    { K40,   K41,   K42,   K43,   K44,   K45,   KC_NO         }, \
+    { K50,   K51,   K52,   K53,   K54,   K55,   K56           }, \
+    { K60,   K61,   K62,   K63,   K64,   K65,   K66           }, \
+    { KC_NO, K71,   K72,   K73,   K74,   K75,   K76           }, \
+	{ K80,   K81,   KC_NO, KC_NO, KC_NO, KC_NO, KC_NO         }, \
+    { KC_NO, KC_NO, KC_NO, K93,   KC_NO, K95,   K96           } \
+}
```
Comment (noroadsleft) [inline keyboards/idb/idb_60/idb_60.h]: Would you please use the following block for Lines 42-59? No logic changes, just cleans up the line indenting. (The other layout macro above could use a similar change.)

```c
#define LAYOUT_default( \
    K00, K10, K01, K11, K02, K12, K03, K13, K04, K14, K05, K15, K06, K16,\
    K20, K30, K21, K31, K22, K32, K23, K33, K24, K34, K25, K35, K26, K36,\
    K40, K50, K41, K51, K42, K52, K43, K53, K44, K54, K45, K55,      K56,\
    K60,      K61, K71, K62, K72, K63, K73, K64, K74, K65, K75, K66, K76,\
    K80,      K81,                     K93,                K95,      K96\
) { \
    { K00,   K01,   K02,   K03,   K04,   K05,   K06,    KC_NO }, \
    { K10,   K11,   K12,   K13,   K14,   K15,   K16           }, \
    { K20,   K21,   K22,   K23,   K24,   K25,   K26           }, \
    { K30,   K31,   K32,   K33,   K34,   K35,   K36           }, \
    { K40,   K41,   K42,   K43,   K44,   K45,   KC_NO         }, \
    { K50,   K51,   K52,   K53,   K54,   K55,   K56           }, \
    { K60,   K61,   K62,   K63,   K64,   K65,   K66           }, \
    { KC_NO, K71,   K72,   K73,   K74,   K75,   K76           }, \
    { K80,   K81,   KC_NO, KC_NO, KC_NO, KC_NO, KC_NO         }, \
    { KC_NO, KC_NO, KC_NO, K93,   KC_NO, K95,   K96           } \
}
```

File: keyboards/idb/idb_60/info.json [hunk 18]
```diff
@@ -1,12 +1,73 @@
...
-  "keyboard_name": "idb_60",
+  "keyboard_name": "idb 60",
   "url": "https://idb-keyboards.xyz/60",
   "maintainer": "pngu",
   "width": 15,
   "height": 5,
   "layouts": {
     "LAYOUT": {
-      "layout": [
+	   "LAYOUT_default": [
```
Comment (noroadsleft) [inline keyboards/idb/idb_60/info.json]: This file could use some line indent fixes too.

File: keyboards/idb/idb_60/readme.md [hunk 19]
```diff
@@ -8,10 +8,19 @@ Hardware Availability: [/u/omgitspngu](https://www.reddit.com/user/omgitspngu/)
 
 Make example for this keyboard (after setting up your build environment):
 
-    make idb_60:default
+    make idb/idb_60:default
 
 Or to make and flash:
 
-    make idb_60:default:dfu
+    make idb/idb_60:default:dfu
```
Comment (noroadsleft) [inline keyboards/idb/idb_60/readme.md]: The `:flash` target will call the correct program based on the defined bootloader, and supports the most common bootloaders used within QMK.

```suggestion
    make idb/idb_60:default:flash
```

**Comment by noroadsleft** [comment]
@drashna Can you check out the LUFA changes here? I'm not sure if it's messed up on itspngu's end or mine at this point.

File: keyboards/idb/idb_60/config.h [hunk 20]
```diff
@@ -48,7 +48,8 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
 /* COL2ROW or ROW2COL */
 #define DIODE_DIRECTION COL2ROW
 
-/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+/* Debounce reduces chatter (unintended double-presses) - see https://docs.qmk.fm/#/feature_debounce_type for more information */
+#define DEBOUNCE_TYPE eager_pk
```
Comment (drashna) [inline keyboards/idb/idb_60/config.h]:
```suggestion
```

This needs to be set in the rules.mk file, not here.  And as:
```make
DEBOUNCE_TYPE = eager_pk
```

File: keyboards/idb/idb_60/rules.mk (hunk 13 above)
Comment (drashna) [reply keyboards/idb/idb_60/rules.mk]: This one doesn't appear to be updated, yet.

File: keyboards/idb/idb_60/rules.mk [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+EXTRAKEY_ENABLE = yes			# Audio control and System control
+CONSOLE_ENABLE = no				# Console for debug
+COMMAND_ENABLE = no				# Commands for debug and configuration
+KEYBOARD_LOCK_ENABLE = no		# Allow locking of keyboard via magic key
+SLEEP_LED_ENABLE = no			# Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes				# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no			# Enable keyboard backlight functionality
+MIDI_ENABLE = no				# MIDI controls
+UNICODE_ENABLE = no				# Unicode
+BLUETOOTH_ENABLE = no			# Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (drashna) [inline keyboards/idb/idb_60/rules.mk]: For the debouncing, you want this:

```suggestion
BLUETOOTH_ENABLE = no			# Enable Bluetooth with the Adafruit EZ-Key HID

DEBOUNCE_TYPE = eager_pk
```

**Comment by itspngu** [comment]
> 
> 
> Also, it looks like you have updated the lufa submodule. You may need to run `make git-submodule` and add a commit, so it reverts the changes.

I don't know what I did anymore but I didn't touch any of the lufa files themselves. My branch was stale (from somewhen around march or april) and after updating it things didn't compile anymore with error messages suggesting something about Lufa missing. I then googled a bit and tried a series of git submodule commands until it eventually worked.

This is the output of `make git-submodule`:

`root@DESKTOP-QQ80MAV:/mnt/d/keyboards/qmk_idb60/qmk_firmware# make git-submodule
QMK Firmware 0.7.106
Synchronizing submodule url for 'lib/chibios'
Synchronizing submodule url for 'lib/chibios-contrib'
Synchronizing submodule url for 'lib/googletest'
Synchronizing submodule url for 'lib/lufa'
Synchronizing submodule url for 'lib/ugfx'
usage: git submodule [--quiet] add [-b <branch>] [-f|--force] [--name <name>] [--reference <repository>] [--] <repository> [<path>]
   or: git submodule [--quiet] status [--cached] [--recursive] [--] [<path>...]
   or: git submodule [--quiet] init [--] [<path>...]
   or: git submodule [--quiet] deinit [-f|--force] [--] <path>...
   or: git submodule [--quiet] update [--init] [--remote] [-N|--no-fetch] [-f|--force] [--checkout|--merge|--rebase] [--reference <repository>] [--recursive] [--] [<path>...]
   or: git submodule [--quiet] summary [--cached|--files] [--summary-limit <n>] [commit] [--] [<path>...]
   or: git submodule [--quiet] foreach [--recursive] <command>
   or: git submodule [--quiet] sync [--recursive] [--] [<path>...]
Makefile:616: recipe for target 'git-submodule' failed
make: *** [git-submodule] Error 1`

I have no clue what's going on anymore. I tried sync, update, init, etc, also removed the lib/lufa folder as a whole as a last ditch and it seems the Lufa submodule still does not point to the commit it should point to:

`root@DESKTOP-QQ80MAV:/mnt/d/keyboards/qmk_idb60/qmk_firmware# git submodule status
 587968d6cbc2b0e1c7147540872f2a67e59ca18b lib/chibios (remotes/origin/dynamic_sdu)
 ede48346eee4b8d6847c19bc01420bee76a5e486 lib/chibios-contrib (remotes/origin/17.6.3_fixes-4-gede4834)
 ec44c6c1675c25b9827aacd08c02433cccde7780 lib/googletest (release-1.8.0)
-4f66a7a3258bae4d2a859529a719bb5142506346 lib/lufa
 3e97b74e03c93631cdd3ddb2ce43b963fdce19b2 lib/ugfx (v2.7-9-g3e97b74)`

If you guys don't know what's up either (I appreciate the help) I guess I'll have to resort to trying to get a fresh fork of the repo and try things from there again.

**Comment by itspngu** [comment]
Okay, I'm positive everything's fit now. Thanks to Curry and tz on the QMK discord for walking me through it!

**Comment by Wilba6582** [comment]
👍

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**